pimd/pim_cmd.c

```diff
@@ -5413,28 +4575,43 @@ DEFUN (show_ip_pim_rp,
        "Multicast Group range\n"
        JSON_STR)
 {
-	int idx = 2;
-	struct vrf *vrf = pim_cmd_lookup_vrf(vty, argv, argc, &idx);
-	bool uj = use_json(argc, argv);
+	struct pim_instance *pim;
+	struct vrf *v;
+	json_object *json_parent = NULL;
 	struct prefix *range = NULL;
 
-	if (!vrf)
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!v)
+		return CMD_WARNING;
+
+	pim = pim_get_pim_instance(v->vrf_id);
+
+	if (!pim) {
+		vty_out(vty, "%% Unable to find pim instance\n");
 		return CMD_WARNING;
+	}
 
-	if (argv_find(argv, argc, "A.B.C.D/M", &idx)) {
+	if (group_str) {
 		range = prefix_new();
-		(void)str2prefix(argv[idx]->arg, range);
+		memcpy(range, &group, sizeof(group));
```

Comment:
To copy different `struct prefix_*` to `struct prefix`, `prefix_copy()` should be used to avoid mistakes/typos.

(For the same `struct prefix_*`, simple assignment should be used. ⇒ never memcpy)